src/Simplify.elm

```diff
@@ -1385,6 +1391,93 @@ expressionVisitorHelp node context =
             onlyErrors []
 
 
+distributeFieldAccess : String -> Node Expression -> Node String -> List (Error {})
+distributeFieldAccess kind ((Node recordRange _) as record) (Node fieldRange fieldName) =
+    case recordLeavesRanges record of
+        { records, withoutParens, withParens } ->
+            if List.length withParens > 1 then
+                -- Never introduce more than one set of parentheses
+                -- This is an heuristic for avoiding distributing field access
+                -- when the expressions are not amenable to simplification
+                -- while still letting it run when there is a single exception
+                -- or if it's a let/in
```

Comment:
I think it would be better if following what Simon Lydell said: Only apply the transformation if we see at least one place where we see a simplification to apply. I think we could do the same checks that we do in `recordAccessChecks`, and if we have at least one branch where it matches, then we report an error, otherwise we don't.

That sounds like a more useful approach than this heuristic, where people will have trouble figuring out why the optimization didn't kick in.

Reply:
So `if List.isEmpty records`? Makes sense. And then a boolean to do it anyway for `let`s?

Reply:
Pushed